src/main/java/org/jabref/gui/util/DefaultTaskExecutor.java

```diff
@@ -92,19 +93,20 @@ public static void runInJavaFXThread(Runnable runnable) {
     }
 
     @Override
-    public <V> Future<V> execute(BackgroundTask<V> task) {
+    public <V> CompletableFuture<V> execute(BackgroundTask<V> task) {
         return execute(getJavaFXTask(task));
     }
 
     @Override
-    public <V> Future<V> execute(Task<V> task) {
-        executor.submit(task);
-        return task;
+    public <V> CompletableFuture<V> execute(Task<V> task) {
+        CompletableFuture<V> future = new CompletableFuture<>();
+        executor.submit((Callable<V>) task::get);
+        return future;
     }
 
     @Override
-    public <V> Future<?> schedule(BackgroundTask<V> task, long delay, TimeUnit unit) {
-        return scheduledExecutor.schedule(getJavaFXTask(task), delay, unit);
+    public <V> Future<V> schedule(BackgroundTask<V> task, long delay, TimeUnit unit) {
+        return scheduledExecutor.schedule(() -> getJavaFXTask(task).get(), delay, unit);
```

Comment:
As above, this is not equivalent. You really need to submit the task, otherwise it doesn't trigger the handlers.